### PR TITLE
Bastion MachineDeployment for CAPZ

### PR DESCRIPTION
**Pull request by primeroz**
towards https://github.com/giantswarm/giantswarm/issues/25087

Setup Bastion Host 

* one per cluster, no matter if MC or WC 
* exposes the bastion publicly when `network.mode` is not private

### CODE REVIEW

File: helm/cluster-azure/values.yaml [hunk 1]
```diff
@@ -18,6 +18,12 @@ network:
   serviceCIDR: 172.31.0.0/16
   podCIDR: "192.168.0.0/16"
   hostCIDR: "10.0.0.0/8"
+  mode: public
+
+bastion:
+  enabled: true
+  instanceType: Standard_D2s_v5
+  replicas: 1
```
Comment (bavarianbidi) [inline helm/cluster-azure/values.yaml:26]:
```suggestion
```

i would propose to have a fixed number of `1` and not make it configurable through the helm chart

File: helm/cluster-azure/files/etc/ssh/sshd_config_bastion [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+# Non defaults (#100)
+ClientAliveCountMax 2
+PasswordAuthentication no
+TrustedUserCAKeys /etc/ssh/trusted-user-ca-keys.pem
+MaxAuthTries 5
+LoginGraceTime 60
+AllowTcpForwarding yes
+AllowAgentForwarding yes
+CASignatureAlgorithms ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521,ssh-ed25519,rsa-sha2-512,rsa-sha2-256,ssh-rsa
+Port 30000
```
Comment (bavarianbidi) [inline helm/cluster-azure/files/etc/ssh/sshd_config_bastion:17]: why Port `30000`?

File: helm/cluster-azure/values.yaml (hunk 1 above)
Comment (primeroz) [reply helm/cluster-azure/values.yaml:26]: :+1: will do that

**Review by bavarianbidi — APPROVED** [review]

File: helm/cluster-azure/values.schema.json [hunk 3]
```diff
@@ -35,6 +35,30 @@
...
+        "clusterDescription": {
+            "description": "User-friendly description of the cluster's purpose.",
+            "title": "Cluster description",
+            "type": "string"
+        },
+        "clusterName": {
+            "description": "Unique identifier, cannot be changed after creation.",
+            "title": "Cluster name",
+            "type": "string"
+        },
```
Comment (marians) [inline helm/cluster-azure/values.schema.json:61]: Just noticed that this PR added clusterDescription and clusterName back, which have been moved to /metadata in a different PR.

Comment (marians) [reply helm/cluster-azure/values.schema.json:61]: I'll make a PR to remove it.

Comment (marians) [reply helm/cluster-azure/values.schema.json:61]: https://github.com/giantswarm/cluster-azure/pull/48

Comment (primeroz) [reply helm/cluster-azure/values.schema.json:61]: oh sorry ... i thought i addressed it :+1:

**Pull request merged**